Adiciona docs de contribuição (#151)
SOS-RS/backend: CONTRIBUTING.md

# Contribuindo

Você pode contribuir com o projeto de diversas formas, implementando uma
funcionalidade nova, corrigindo um bug, procurando bugs, revisando pull
requests, entre outras.
Para se inteirar do projeto, entre no
[Discord](https://discord.gg/vjZS6BQXvM) e participe das discussões.

## 🤝 Contribuindo com atividades que não são de código

> O projeto precisa de ajuda em diversas frentes diferentes, como QA, produto,
> design e gestão. Entre no servidor do Discord, onde há canais específicos para
> essas atividades.

Reviewer: ```suggestion
O projeto precisa de ajuda em diversas frentes diferentes como: QA, produto,
design e gestão. Entre no servidor do Discord onde há canais específicos para
essas atividades.
```

Se você encontrou um bug, vá nas
[issues](https://github.com/SOS-RS/backend/issues)
do projeto e reporte-o lá. Verifique antes se já não existe um bug aberto para o

> problema que quer relatar, usando a busca. O mesmo vale para novas
> funcionalidades.

Reviewer: ```suggestion
problema que quer relatar, usando a busca. O mesmo vale para novas
funcionalidades quando você for abrir um pr que não está relacionado diretamente a uma issue aberta.
```

Author: Nesse caso, não está seguindo no mesmo assunto, o parágrafo fala de novos itens sendo reportados, não de PRs.

Reviewer: Ah beleza, eu entendi errado 🚀

O restante deste documento focará nas contribuições de código.

## ✅ Escolhendo qual será sua contribuição de código

Verifique no [projeto do Github](https://github.com/orgs/SOS-RS/projects/1)
quais funcionalidades ainda não foram implementadas e já estão prontas para

> serem desenvolvidas, elas estarão na coluna "Disponível pra dev". Lá há itens de
> backend e frontend, então atente-se para qual você gostaria de participar.

Reviewer: ```suggestion
serem desenvolvidas, elas estarão na coluna "Disponível pra dev". Nesse projeto existem itens de
`backend` e `frontend`, então atente-se para qual você gostaria de participar.
```

Author: Não acho que precisamos marcar backend e frontend no formato de código. Poderia ser itálico, por ser lingua estrangeira, mas não estamos fazendo isso em outros lugares...

Após escolher o item que quer trabalhar, faça um comentário no issue informando
que quer contribuir para sua entrega. Uma pessoa que administra o repositório
marcará você como a pessoa responsável por aquele issue, e marcará o item como
"Em desenvolvimento".

A partir daí você já pode trabalhar no item que escolheu.

> Você também pode mandar a contribuição diretamente, sem avisar, mas corre o
> risco de alguém solicitar para trabalhar no item e entregá-lo junto ou antes de
> você, desperdiçando esforços. Faça isso só se a correção for bem rápida, para
> evitar o desperdício.

Reviewer: ```suggestion
Você também pode mandar a contribuição diretamente sem avisar, mas corre o
risco de alguém solicitar para trabalhar no item e entregá-lo junto ou antes de
você, desperdiçando assim esforços. Somente faça isso se a correção for bem rápida e pontual para
evitar o desperdício.
```

⚠️ **Importante**: Itens de alta prioridade precisam ser entregues o mais rápido possível,
idealmente em até dois dias. Verifique se tem tempo livre suficiente para se
dedicar a um item de urgência, a fim de evitar segurar o item por tempo demais

> de forma desnecessária.

Reviewer 2: será que aqui compensa a gente definir um sla por tamanho de task urgente? 
ex:
P - 1 dia
M - 2 dias
G - 3 dias

Author: Em conversa com o @rhuam ficou definido 2 dias pra itens urgentes, inicialmente. @larissapissurno bora ter essa conversa no Discord? Aí fica mais fácil de ouvir todo mundo, também.

Reviewer: Acho um pouco mais delicado esse assunto de P/M/G visto que estamos com projeto aberto e qualquer pessoa deveria/poderia ajudar, não seria melhor deixar um texto mais genérico focando no ponto de `itens urgentes deverão ser concluídos o mais rápido possível, então tenha certeza que você possui o tempo necessário para contribuir e se dedicar a tarefa` algo nesse estilo talvez, oq acham?

Author: Em discussão no Discord, foi entendido que não. O problema é alguém pegar um item urgente e levar semanas pra terminar. O item fica travado, quando outra pessoa poderia pegar e entregar mais rápido.

Reviewer 2: opa, concordo @giggio, posso abrir um tópico hoje a noite caso ainda nao exista um 🤝

Reviewer: Meu ponto não era nem esse, obviamente eu concordo com vcs, porém o que está escrito aqui é apenas um combinado, não vejo muito ganho em definir tempo baseado em estimativas P/M/G, pq no fim o resultado vai ser o mesmo, independente do que estiver escrito aqui 🚀
